Gridy/Controller/StartViewController.swift

```diff
@@ -18,7 +19,148 @@ class StartViewController: UIViewController {
+        }
+      } else {
+        troubleAlert(message: "Something went terribly wrong. Gridy is unable to access your \(media) at this time")
+      }
+    }
+    
+    /// Ask the user the authorization to use the camera.
+    /// User can redirect to Settings if authorization not granted.
+    /// Call Image Picker if authorization has already been granted
+    ///
+    /// - Parameters:
+    ///   - status: Authorization status to use Camera
+    ///   - noPermissionMessage: Message to display if authorization not granted
+    func actionAccordingTo(status: AVAuthorizationStatus , noPermissionMessage: String?) {
+        let sourceType = UIImagePickerController.SourceType.camera
+        switch status {
+        case .notDetermined:
+            AVCaptureDevice.requestAccess(for: AVMediaType.video, completionHandler: { (granted) in
+                if granted {
+                    self.presentImagePicker(sourceType: sourceType)
+                    print ("Access authorized.")
+                } else {
+                    self.troubleAlert(message: noPermissionMessage)
+                }
+            })
+        case .authorized:
+            self.presentImagePicker(sourceType: sourceType)
+            print ("Access authorized.")
+        case .denied, .restricted:
+            self.troubleAlert(message: noPermissionMessage)
+        default:
+            self.troubleAlert(message: "Something went terribly wrong. Gridy is unable to access your photo library at this time.")
+        }
+    }
+    
+    /// Ask the user the authorization to use the photo library.
+    /// User can redirect to Settings if authorization not granted.
+    /// Call Image Picker if authorization has already been granted
+    ///
+    /// - Parameters:
+    ///   - status: Authorization status to use PhotoLibrary
+    ///   - noPermissionMessage: Message to display if authorization not granted
+    func actionAccordingTo(status: PHAuthorizationStatus , noPermissionMessage: String?) {
+        let sourceType = UIImagePickerController.SourceType.photoLibrary
+        switch status {
+        case .notDetermined:
+            PHPhotoLibrary.requestAuthorization({ (newStatus) in
+                if newStatus == .authorized {
+                    self.presentImagePicker(sourceType: sourceType)
+                    print ("Access authorized.")
+                }
+                else {
+                    self.troubleAlert(message: noPermissionMessage)
+                }
+            })
+        case .authorized:
+            self.presentImagePicker(sourceType: sourceType)
+            print ("Access authorized.")
+        case .denied, .restricted:
+            self.troubleAlert(message: noPermissionMessage)
+        default:
+            self.troubleAlert(message: "Something went terribly wrong. Gridy is unable to access your photo library at this time.")
+        }
+    }
```

Comment:
Regarding the last mentoring session on this subject, I can confirm that the 2 above methods are not easy to factorize without using an abstract service that would allow you to handle `PhotoLibrary` and `Camera` as if they were the same thing.